pcdsdevices/ipm.py

```diff
@@ -25,14 +29,57 @@ class IPMTarget(InOutRecordPositioner):
     _transmission = {st: 0.8 for st in in_states}
 
 
-class IPM(Device):
+class IPMDiode(Device):
+    """
+    Diode of a standard intensity position monitor.
+
+    This device has members for x and y motors, where x-motion is set up as an
+    IMS motor and y-motion is an InOutRecordPositioner that moves the diode
+    position to any of the four set positions, or out. There is also a member,
+    y, which points to the motor of the y-motion.
+    """
+
+    x = FCpt(IMS, '{self.x_prefix}', kind='normal')
+    state = Cpt(InOutRecordPositioner, '', kind='normal')
+
+    def __init__(self, prefix, *, name,  **kwargs):
+        self.x_prefix = 'XMOTOR'
+        super().__init__(prefix, name=name, **kwargs)
+        self.y = self.state.motor
+
+    @property
+    def inserted(self):
+        """Returns ``True`` if diode is inserted."""
+        return self.state.inserted
+
+    @property
+    def removed(self):
+        """Returns ``True`` if diode is removed."""
+        return self.state.removed
+
+    def insert(self, moved_cb=None, timeout=None, wait=False):
+        """Moves the diode into the beam."""
+        return self.state.insert(moved_cb=moved_cb, timeout=timeout,
+                                 wait=wait)
+
+    def remove(self, moved_cb=None, timeout=None, wait=False):
+        """Moves the diode out of the beam."""
+        return self.state.remove(moved_cb=moved_cb, timeout=timeout,
+                                 wait=wait)
+
+    remove.__doc__ += insert_remove
+
+    tab_whitelist = ['target', 'diode']
```

Comment:
settings like this are best kept near the top of the class def right under the component definitions so they are easy to find